src/main/java/com/aws/greengrass/logmanager/LogManagerService.java

```diff
@@ -649,52 +639,38 @@ private void processLogsAndUpload() throws InterruptedException {
     }
 
     /**
-     * calculate the hash for the active file, which takes and digests first some lines (default is 1) in the file.
-     * If the content of file is insufficient, or incorrect setting for DEFAULT_LINES_FOR_DIGEST_NUM, the hash is null.
-     * @param activeFile
+     * calculate the hash for the log file, which takes and digests first some lines (default is 1) in the file.
+     * If the content of file is insufficient, the hash is null.
+     * @param logFile
      * @return the nullable calculated hash for active file
      */
-    private Optional<String> calculateHashForActiveFile(File activeFile) {
-        Optional activeFileHash = Optional.ofNullable(null);
-        // String activeFileHash = null;
-        try (SeekableByteChannel chan = Files.newByteChannel(activeFile.toPath(), StandardOpenOption.READ);
-             PositionTrackingBufferedReader r =
-                new PositionTrackingBufferedReader(new InputStreamReader(Channels.newInputStream(chan),
-                        StandardCharsets.UTF_8))) {
-            r.setInitialPosition(0);
-            chan.position(0);
+    private Optional<String> calculateHashForLogFile(File logFile) {
+        Optional fileHash = Optional.ofNullable(null);
+        try (BufferedReader r = Files.newBufferedReader(logFile.toPath(), StandardCharsets.UTF_8)) {
             StringBuilder data = new StringBuilder(r.readLine());
```

Comment:
Why not start the string builder empty. The while loop should handle the `r.readLine()`

```
StringBuilder data = new StringBuilder();
```